mne/io/eyelink/_utils.py

```diff
@@ -187,8 +187,14 @@ def _get_recording_datetime(fname):
                     # Eyelink measdate timestamps are timezone naive.
                     # Force datetime to be in UTC.
                     # Even though dt is probably in local time zone.
-                    dt_naive = datetime.strptime(dt_str, fmt)
-                    return dt_naive.replace(tzinfo=tz)  # make it dt aware
+                    try:
+                        dt_naive = datetime.strptime(dt_str, fmt)
+                        return dt_naive.replace(tzinfo=tz)  # make it dt aware
+                    except ValueError:
+                        # date string is missing or in an unexpected format
+                        logger.debug("Could not detect date in file.")
+                        return
```

Comment:
Typically you want as few lines as possible in there to make it clear what can actually fail, so probably this is better:
```suggestion
                    try:
                        dt_naive = datetime.strptime(dt_str, fmt)
                    except ValueError:
                        # date string is missing or in an unexpected format
                        logger.info(f"Could not detect date from file with date entry: {repr(dt_str)}")
                        return
                    else:
                        return dt_naive.replace(tzinfo=tz)  # make it dt aware 
```
Also I changed it to `info` since users should probably see something if there's a DATE in the file but it can't be parsed I think?